sfn-query-large-datasets-cdk/app.py

```diff
@@ -0,0 +1,307 @@
+#!/usr/bin/env python3
+import json
+import random
+import string
+import os
+from aws_cdk import (
+    App,
+    Aspects,
+    Environment,
+    RemovalPolicy,
+    Stack,
+    aws_s3 as s3,
+    aws_iam as iam,
+    aws_sns as sns,
+    aws_kms as kms,
+    aws_glue as glue,
+    aws_athena as athena,
+    aws_stepfunctions as sfn,
+    aws_logs as logs,
+    CfnOutput as outputs
+)
+from constructs import Construct
+from cdk_nag import AwsSolutionsChecks, NagSuppressions
+
+
+class SfnQueryLargeDatasetsCdkStack(Stack):
+    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
+        super().__init__(scope, construct_id, **kwargs)
+
+        # Random string generation for suffixes
+        random_string = "".join(random.choices(string.ascii_lowercase, k=8))
+
+        # S3 Bucket
+        s3_bucket = s3.Bucket(self, "s3_query_large_datasets", 
+            bucket_name=f"stepfunctions-query-large-datasets-{random_string}",
+            removal_policy=RemovalPolicy.DESTROY,
+            auto_delete_objects=True,
+            block_public_access=s3.BlockPublicAccess.BLOCK_ALL,
+            enforce_ssl=True,
+        )
+
+        # SNS KMS key
+        sns_key = kms.Alias.from_alias_name(self, "SnsKey", "alias/aws/sns")
+
+        # SNS Topic
+        athena_updates_topic = sns.Topic(self, "athena_updates",
+            topic_name="athena-updates-topic",
```

Comment:
If we use the `random_string` for other resource names, it'd make sense to use it also here.